### PR TITLE
New date picker and aria-labels for IconButtons

### PR DESCRIPTION
**Pull request by jolle**
#### Summary

- Added mandatory aria-label field to IconButtons
- Replaced all existing date pickers with react-aria's date picker with customised design

### CODE REVIEW

File: frontend/src/e2e-test/pages/employee/holiday-periods.ts [hunk 1]
```diff
@@ -89,19 +104,20 @@ export class HolidayPeriodsPage {
...
     descriptionLinkEn?: string
-    activeStart?: string
-    activeEnd?: string
+    activeRange?: FiniteDateRange
     fixedPeriodOptions?: string
     fixedPeriodOptionLabel?: string
   }) {
     for (const [key, val] of Object.entries(params)) {
       if (val !== undefined) {
         if (key === 'period') {
```
Comment (akheron) [inline frontend/src/e2e-test/pages/employee/holiday-periods.ts:113]: `params` doesn't have `period`, so is this special handling needed? 🤔

File: frontend/src/lib-components/molecules/date-picker/Calendar.tsx [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    <td {...cellProps}>
+      <Cell
+        {...buttonProps}
+        ref={ref}
+        hidden={isOutsideVisibleRange}
+        className={classNames(
+          isSelected && 'selected',
+          isDisabled && 'disabled',
+          isUnavailable && 'unavailable'
+        )}
```
Comment (akheron) [inline frontend/src/lib-components/molecules/date-picker/Calendar.tsx:195]: Alternative syntax which I find more readable:
```
className={classNames({
  selected: isSelected,
  disabled: isDisabled,
  unavailable: isUnavailable
})}
```

File: frontend/src/lib-components/molecules/date-picker/DateField.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  state: DateFieldState
+}) {
+  const ref = React.useRef<HTMLDivElement>(null)
+  const { segmentProps } = useDateSegment(segment, state, ref)
+
+  return (
+    <Segment
+      {...segmentProps}
+      ref={ref}
+      className={classNames(segment.isPlaceholder && 'placeholder')}
```
Comment (akheron) [inline frontend/src/lib-components/molecules/date-picker/DateField.tsx:88]: Ditto for `classNames`. And seems there are even more in other files.

**Review by akheron — CHANGES_REQUESTED** [review]

**Comment by akheron** [comment]
Quite a PR 🏅

File: frontend/src/e2e-test/pages/employee/holiday-periods.ts (hunk 1 above)
Comment (jolle) [reply frontend/src/e2e-test/pages/employee/holiday-periods.ts:113]: I don't think it's needed anymore, the field was removed in 4bfa5920cb1fade8b6ac32a15742c61ff656ccce.

File: frontend/src/lib-components/atoms/Popover.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  openAbove?: boolean
+} & AriaDialogProps
+
+const OverlayBox = styled.div`
+  background-color: ${(p) => p.theme.colors.grayscale.g0};
+  box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.25);
+  border-radius: 4px;
+`
+
+const OverlayContainer = styled.div<{ openAbove?: boolean }>`
```
Comment (akheron) [inline frontend/src/lib-components/atoms/Popover.tsx:29]:
```suggestion
const OverlayContainer = styled.div<{ $openAbove?: boolean }>`
```

When prefixed with a `$`, the attribute won't end up in the dom.

**Review by akheron — APPROVED** [review]

**Comment by akheron** [comment]
Pending UX approve, but the code looks good 👍

**Pull request merged**